### ISSUE TITLE
dials.index: if iterating through many experiments please give hint of experiment number

### ISSUE THREAD

**Issue by graeme-winter** [dials/dials#2450]
As in spot finding, where we say "finding spots on imageset N"

Indexing experiment N would seem polite (I am indexing a 200-sweep data set)

**Comment by dagewa** [comment]
@graeme-winter can you please try the branch https://github.com/dials/dials/tree/2450-dialsindex-if-iterating-through-many-experiments-please-give-hint-of-experiment-number to see if this does what you want? I am not in a position to test at the moment.

**Comment by graeme-winter** [comment]
> @graeme-winter can you please try the branch https://github.com/dials/dials/tree/2450-dialsindex-if-iterating-through-many-experiments-please-give-hint-of-experiment-number to see if this does what you want? I am not in a position to test at the moment.

Wilco, thank you (got bogged down in other issues)

**Comment by graeme-winter** [comment]
Mechanism:

```
dials.import <many data>
dials.find_spots imported.expt
dials.index imported.expt strong.refl joint=0
```

**Comment by dagewa** [comment]
I think the branch does _not_ do what you want. I imported 4 datasets, and in each case looked for multiple lattices. The printed experiment id resets after each imageset:
```
$ grep "Adding indexed" dials.index.log 
Adding indexed experiment 0
Adding indexed experiment 1
Adding indexed experiment 0
Adding indexed experiment 1
Adding indexed experiment 2
Adding indexed experiment 0
Adding indexed experiment 0
Adding indexed experiment 1
Adding indexed experiment 2

```

**Comment by dagewa** [comment]
Had a look at this again but forgot what data I was looking at before. I'm not sure we have a suitable entry in `dials-data`, but I think the issue can be explored by splitting a dataset over 9 sweeps like this:
```
dials.import $(dials.data get -q x4wide)/X4_wide_M1S4_2_00{0..8}{1..8}.cbf 
dials.find_spots imported.expt
dials.index imported.expt strong.refl joint_indexing=False
```

**Comment by ndevenish** [comment]
Closed by #2470
